cmd/kube-apiserver/app/server.go

```diff
@@ -279,104 +192,34 @@ func Run(s *options.APIServer) error {
 		glog.Errorf("Failed to create clientset: %v", err)
 	}
 
-	legacyV1Group, err := registered.Group(api.GroupName)
-	if err != nil {
-		return err
-	}
-
-	storageDestinations := genericapiserver.NewStorageDestinations()
-
-	storageVersions := s.StorageGroupsToGroupVersions()
-	if _, found := storageVersions[legacyV1Group.GroupVersion.Group]; !found {
-		glog.Fatalf("Couldn't find the storage version for group: %q in storageVersions: %v", legacyV1Group.GroupVersion.Group, storageVersions)
-	}
-	etcdStorage, err := newEtcd(api.Codecs, storageVersions[legacyV1Group.GroupVersion.Group], "/__internal", s.EtcdConfig)
-	if err != nil {
-		glog.Fatalf("Invalid storage version or misconfigured etcd: %v", err)
+	resourceEncoding := genericapiserver.NewDefaultResourceEncodingConfig()
```

Comment:
Pointing out a change from before: before we were ensuring that s.StorageGroupsToGroupVersion() contained all groups. Now resourceEncoding.StorageEncodingFor() silently returns the default version for that group, even if it was not specified in s.StorageGroupsToGroupVersion().

If we intend to use StorageGroupsToGroupVersion() only in cases of overrides, then we should probably rename it and update the comment?
Else resourceEncoding.StorageEncodingFor() should return an error if group was not found.